FWCore/MessageLogger/src/ELseverityLevel.cc

```diff
@@ -122,148 +123,80 @@ namespace edm {
 
   int ELseverityLevel::getLevel() const { return myLevel; }
 
-  const ELstring ELseverityLevel::getSymbol() const {
-    ELstring result;
-
-    switch (myLevel) {
-      default:
-        result = "0";
-        break;
-      case ELsev_zeroSeverity:
-        result = "--";
-        break;
-      case ELsev_success:
-        result = "-d";
-        break;  // 4/21/09 mf
-      case ELsev_info:
-        result = "-i";
-        break;
-      case ELsev_warning:
-        result = "-w";
-        break;
-      case ELsev_error:
-        result = "-e";
-        break;
-      case ELsev_unspecified:
-        result = "??";
-        break;
-      case ELsev_severe:
-        result = "-s";
-        break;
-      case ELsev_highestSeverity:
-        result = "!!";
-        break;
-    };  // switch
-
-    return result;
+  const std::string& ELseverityLevel::getSymbol() const {
+    static const auto symbols = []() {
```

Comment:
```suggestion
    static constexpr auto symbols = []() constexpr {
```

Reply:
I see now this can't be `constexpr` unless this is also changed to use `std::string_view` throughout.

Reply:
Right, `constexpr` `std::string` constructor comes only with C++20 (not sure if there would be other hurdles). I gave a try with `std::string_view` (also in the `std::map`), but that lead to failure in one MessageLogger unit test.